x-pack/plugins/endpoint/server/services/resolver/search_handler.test.ts

```diff
@@ -0,0 +1,311 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import {
+  elasticsearchServiceMock,
+  loggingServiceMock,
+} from '../../../../../../src/core/server/mocks';
+import { ResolverSearchHandler } from './search_handler';
+import { IScopedClusterClient } from 'kibana/server';
+import { EndpointAppContext, Total } from '../../types';
+import { EndpointConfigSchema } from '../../config';
+import { PaginationInfo, getPagination } from './query_builder';
+import { CountResponse } from 'elasticsearch';
+import { ResolverDataHit, parsePhase0EntityID, buildPhase0EntityID } from './common';
+import { ResolverData, Pagination } from '../../../common/types';
+import { EventBuilder } from './event_builder.test';
+import { Phase0Builder } from './phase0_builder.test';
+import { Phase1Builder } from './phase1_builder.test';
+
+function buildPageInfo(page?: number, pageSize?: number): PaginationInfo {
+  return {
+    page,
+    pageSize,
+  };
+}
+
+function buildCountResponse(count: number, scopedClient: jest.Mocked<IScopedClusterClient>) {
+  scopedClient.callAsCurrentUser.mockImplementationOnce(() =>
+    Promise.resolve({
+      count,
+      _shards: {
+        total: 1,
+        successful: 1,
+        failed: 0,
+        skipped: 0,
+      },
+    } as CountResponse)
+  );
+}
+
+interface BuiltHits {
+  total: number;
+  hits: ResolverDataHit[];
+  children: ResolverData[];
+  origin: ResolverData[];
+}
+
+function buildResolverHits(
+  builder: EventBuilder,
+  numNodes: number,
+  eventsPerNode: number
+): BuiltHits {
+  const hits: ResolverDataHit[] = [];
+  const children: ResolverData[] = [];
+  const origin: ResolverData[] = [];
+  // start after the entity id so there aren't any collisions
+  let nodeIter = builder.startingChildrenEntityID();
+  for (; nodeIter < numNodes + builder.startingChildrenEntityID(); nodeIter++) {
```

Comment:
I think maybe I get it now, are you trying to manually paginate this? If so, instead you should just do it directly in elasticsearch with something like a `sort` value on the timestamp and a `search_after` where the `search_after` parameter is optional based off of a query parameter that is passed.

Reply:
This test class doesn't interact with elastic search except for one place that I mock out the count call, it's just creating a bunch a test data stored in memory.

Reply:
hah, didn't see the `.test.ts` part of the extension >_>

Reply:
Yeah it's kind of annoying, typescript yells at me when I try to put more than 1 class in a file, even a test file, so I had to separate it out into separate files :(